frontend/src/components/user/ATSUserCreateModal.vue

```diff
@@ -0,0 +1,178 @@
+<script setup lang="ts">
+import axios from 'axios';
+import { onMounted, ref } from 'vue';
+import { Button, Column, DataTable, Dialog, useToast } from '@/lib/primevue';
+
+import { Spinner } from '@/components/layout';
+import { atsService } from '@/services';
+import { BasicResponse, Initiative } from '@/utils/enums/application';
+
+import type { Ref } from 'vue';
+import type { ATSUser, Submission } from '@/types';
+
+// Props
+const { submission } = defineProps<{
+  submission: Submission;
+}>();
+
+// Emits
+const emit = defineEmits(['atsUserLink:link']);
+
+// State
+const atsClientNumber: Ref<string> = ref('');
+const loading: Ref<boolean> = ref(false);
+const proponent: Ref<ATSUser | undefined> = ref(undefined);
+const selectedUser: Ref<ATSUser | undefined> = ref(undefined);
+const visible = defineModel<boolean>('visible');
+
+// Actions
+
+const toast = useToast();
+
+async function createATSClient() {
+  selectedUser.value = undefined;
+  try {
+    loading.value = true;
+
+    const data = setEmptyStringsToNullAndOmitNulls({
+      '@type': 'ClientResource',
+      address: {
+        '@type': 'AddressResource',
+        addressLine1: submission.streetAddress,
+        city: submission.locality,
+        provinceCode: submission.province,
+        primaryPhone: submission.contactPhoneNumber,
+        email: submission.contactEmail
+      },
+      firstName: submission.contactFirstName,
+      surName: submission.contactLastName,
+      regionName: Initiative.HOUSING,
+      optOutOfBCStatSurveyInd: BasicResponse.NO.toUpperCase()
+    });
+
+    const response = await atsService.createATSClient(data);
+    if (response.status === 201) {
+      atsClientNumber.value = response.data.clientId;
+      emit('atsUserLink:link', atsClientNumber.value);
+      visible.value = false;
+      toast.success('New client pushed to ATS');
+    } else {
+      toast.error('Error pushing client to ATS');
+    }
+  } catch (error) {
+    if (!axios.isCancel(error)) toast.error('Error pushing client to ATS ' + error);
+  } finally {
+    loading.value = false;
+  }
+}
+
+function setEmptyStringsToNullAndOmitNulls(obj: Record<string, any>): Record<string, any> {
```

Comment:
Please see `setEmptyStringsToNull` in our frontend utils

Reply:
I tried using it, before I wrote my own, for some reason it wasn't correctly setting values to null. But it's working as expected now. I have removed mine. thanks